desafio-07/arlima/go/desafio7.go

```diff
@@ -0,0 +1,137 @@
+// Adriano Roberto de Lima
+// Desafio 07
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+const (
+	buffersize = 65535
+)
+
+// Check é uma função para verificação de erros na maniputação de arquivos
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+// Reverse é uma função para inverter um array de bytes
+func Reverse(s []byte) []byte {
+	words := []byte(s)
+	for i, j := 0, len(words)-1; i < j; i, j = i+1, j-1 {
+		words[i], words[j] = words[j], words[i]
+	}
+	return []byte(words)
+}
+
+// ProcessBlock é uma função para ler um bloco do dispositivo de entrada e gravar a saida em outro dispositivo.
+// Lê o bloco do fim para o começo e a cada \n encontrado manda para o dispositivo de saida
+// na ordem correta
+func ProcessBlock(f io.ReaderAt, out io.Writer, buffersize int64, position int64, ignorefistreturn bool, line []byte) []byte {
+	buffer := make([]byte, buffersize)
+	_, err := f.ReadAt(buffer, position)
+	check(err)
+	for i := buffersize - 1; i >= 0; i-- {
+		letter := buffer[i]
+		if string(letter) == "\n" {
+			rline := Reverse(line)
+			if !(ignorefistreturn && (i == buffersize-1)) {
+				rline = append(rline, '\n')
+			}
+			_, err := out.Write(rline)
+			check(err)
+			line = []byte{}
+		} else {
+			line = append(line, letter)
+		}
+	}
+	return line
+}
+
+// GetFileName é uma função para ler o parâmetro passado para o programa como o nome do arquivo a ser processado
+func GetFileName() string {
+	argsWithProg := os.Args
+	if len(argsWithProg) != 2 {
+		fmt.Println("ERRO! Sintaxe: desafio7 \"nomedoarquivo\"")
+		os.Exit(0)
+	}
+	return argsWithProg[1]
+}
+
+// Close é utilizada para fechar um arquivo. É necessária quando usamos um defer
+// no fechamento do arquivo
+func Close(c io.Closer) {
```

Comment:
Essa função não é necessára. Voce pode usar um io.ReadCloser onde a chama e simplesmente fazer x.Close(). Isso vai chamar o Close original. Não é necessário tratar erros no Close(). É "best-effort".

Reply:
Eu não usava essas funções. fazia um defer x.Close() direto. Mas o gometalinter reclama da chamada do close sem o tratamento de erros. Segui as dicas desse artigo aqui
http://blevesearch.com/news/Deferred-Cleanup,-Checking-Errors,-and-Potential-Problems/
e criei essas funções. O gometalinter não reclama mais. comentários ?

Reply:
Removido